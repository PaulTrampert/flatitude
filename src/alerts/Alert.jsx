import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import ColorTypes from '../colors/ColorTypes.js';
import AlertAction from './AlertAction.jsx';

class Alert extends React.Component {

  renderIcon = () => {
    let {type} = this.props;
    let iconClass = 'info';
    switch(type) {
    case 'info':
      iconClass = 'fa-info-circle';
      break;
    case 'warning':
    case 'danger':
      iconClass = 'fa-exclamation-triangle';
      break;
    case 'success':
      iconClass = 'fa-check-circle';
      break;
    default:
      iconClass = 'fa-info-circle';
    }

    return <i className={classnames('fa', iconClass)}></i>;
  }

  render() {
    let {
      id,
      type,
      children,
      className,
      actions
    } = this.props;

    return (
      <div className={classnames(['alert', type, className])}>
        <div className="icon">
          {this.renderIcon()}
        </div>
        <div className="message">
          {children}
        </div>
        {
          actions &&
          actions.map((action, idx) => (
            <AlertAction {...action} key={idx} alertId={id} />
          ))
        }
      </div>
    );
  }
}

Alert.propTypes = {
  id: PropTypes.number,
  type: ColorTypes.isRequired,
  children: PropTypes.node,
  className: PropTypes.string,
  actions: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.node.isRequired,
    onClick: PropTypes.func.isRequired
  }))
};

export default Alert;