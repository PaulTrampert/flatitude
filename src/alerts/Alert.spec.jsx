import React from 'react';
import {shallow} from 'enzyme';
import Alert from './Alert.jsx';

describe('Alert', () => {

  describe('render', () => {
    describe('when actions are passed in', () => {
      let actions;

      beforeEach(() => {
        actions = [
          {
            label: 'herp',
            onClick: jest.fn()
          },
          {
            label: 'derp',
            onClick: jest.fn()
          }
        ];
      });

      it('renders the action links', () => {
        const subject = shallow(<Alert type="info" actions={actions}>hi</Alert>);
        expect(subject).toMatchSnapshot();
      });
    });

    it('renders info style', () => {
      const subject = shallow(<Alert type="info">hi</Alert>);
      expect(subject).toMatchSnapshot();
    });

    it('renders warning style', () => {
      const subject = shallow(<Alert type="warning">hi</Alert>);
      expect(subject).toMatchSnapshot();
    });

    it('renders danger style', () => {
      const subject = shallow(<Alert type="danger">hi</Alert>);
      expect(subject).toMatchSnapshot();
    });

    it('renders success style', () => {
      const subject = shallow(<Alert type="success">hi</Alert>);
      expect(subject).toMatchSnapshot();
    });

    it('renders primary style', () => {
      const subject = shallow(<Alert type="success">hi</Alert>);
      expect(subject).toMatchSnapshot();
    });
  });
});