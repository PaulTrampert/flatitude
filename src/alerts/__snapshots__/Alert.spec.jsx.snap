// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert render renders danger style 1`] = `
<div
  className="alert danger"
>
  <div
    className="icon"
  >
    <i
      className="fa fa-exclamation-triangle"
    />
  </div>
  <div
    className="message"
  >
    hi
  </div>
</div>
`;

exports[`Alert render renders info style 1`] = `
<div
  className="alert info"
>
  <div
    className="icon"
  >
    <i
      className="fa fa-info-circle"
    />
  </div>
  <div
    className="message"
  >
    hi
  </div>
</div>
`;

exports[`Alert render renders primary style 1`] = `
<div
  className="alert success"
>
  <div
    className="icon"
  >
    <i
      className="fa fa-check-circle"
    />
  </div>
  <div
    className="message"
  >
    hi
  </div>
</div>
`;

exports[`Alert render renders success style 1`] = `
<div
  className="alert success"
>
  <div
    className="icon"
  >
    <i
      className="fa fa-check-circle"
    />
  </div>
  <div
    className="message"
  >
    hi
  </div>
</div>
`;

exports[`Alert render renders warning style 1`] = `
<div
  className="alert warning"
>
  <div
    className="icon"
  >
    <i
      className="fa fa-exclamation-triangle"
    />
  </div>
  <div
    className="message"
  >
    hi
  </div>
</div>
`;

exports[`Alert render when actions are passed in renders the action links 1`] = `
<div
  className="alert info"
>
  <div
    className="icon"
  >
    <i
      className="fa fa-info-circle"
    />
  </div>
  <div
    className="message"
  >
    hi
  </div>
  <AlertAction
    key="0"
    label="herp"
    onClick={[MockFunction]}
  />
  <AlertAction
    key="1"
    label="derp"
    onClick={[MockFunction]}
  />
</div>
`;
